src/analysis/special_functions/gamma.lean

```diff
@@ -0,0 +1,587 @@
+/-
+Copyright (c) 2022 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+
+import analysis.special_functions.exponential
+import measure_theory.integral.integral_eq_improper
+import measure_theory.integral.limit_comparison
+import analysis.special_functions.integrals
+
+/-!
+# The Gamma function
+
+In this file we define the Γ function (of a real variable in the range `1 ≤ s`), using the
+definition `Γ(s) = ∫ x in 0..∞, x^(s-1) exp(-x) dx`, and prove that it is continuous and satisfies
+the relation `Γ(s+1) = s Γ(s)`. We also prove that `Γ(n+1) = n!` for `n ∈ ℕ`.
+
+TO DO:
+
+- Extend to the whole real line.
+- Allow complex `s` and prove analyticity.
+
+## Tags
+
+gamma
+-/
+
+noncomputable theory
+open finset filter metric interval_integral set function
+open_locale classical topological_space ennreal filter measure_theory
+
+namespace real
+
+/-- Integrand for the Gamma function integral -/
+def F (s x : ℝ) : ℝ := exp(-x) * x^s
+
+/- We prove some lemmas about F:
+
+- for any `s ≥ 0`, F is continuous of `x ∈ [0,∞)`;
+- for any `s ≥ 1`, the derivative of `F(s, -)` at any `x ∈ [0, ∞)` is what it should be;
+- for any `s ≥ 1`, the derivative of `F(s, -)` is integrable on [0, X] for any X ≥ 0;
+- for any `s ∈ ℝ`, F is `O( exp(-(1/2) * x))` as `x → ∞`.
+-/
+
+lemma cont_F (s : ℝ) (h1: 0 ≤ s) : continuous_on s.F (Ici 0) :=
+continuous_on.mul (continuous_on.exp $ continuous_on.neg $ continuous_on_id)
+  $ continuous_on.rpow_const continuous_on_id $ λ _ _, or.inr h1
```

Comment:
Using dot notation this can be written as
```suggestion
continuous_on_id.neg.exp.mul $ continuous_on_id.rpow_const $ λ _ _, or.inr h1
```
Of course this is really the same code written differently, so it doesn't really matter, but tricks like this can make the code a little more readable by removing repetition, and also save you keystrokes when writing in the first place when you get used to it!
We have a little explainer on this at the tips and tricks page, https://leanprover-community.github.io/tips_and_tricks.html#avoiding-many-nested-parentheses-with-